docs/Compiled Braid Migration.md

```diff
@@ -0,0 +1,254 @@
+This feature allows you to create a specialized version of a generic function that only accepts
+specific types. Since functional components are just functions, we can do this for components too:
+
+```tsx
+import { ComponentProps } from 'react';
+import { Autosuggest } from 'braid-design-system';
+
+// A specialized Autosuggest component that only accepts `string`s
+const StringAutosuggest = Autosuggest<string>;
+type AutosuggestProps = ComponentProps<typeof StringAutosuggest>;
+
+// Equivalent to AutosuggestValue<string>
+type Value = AutosuggestProps['value'];
+
+const autosuggestValue: Value = {
+  text: 'myValue',
+  // value is of type string
+  value: 'foo',
+};
+
+// Error: value must be of type string
+autosuggestValue.value = 123;
+```
+
+This has the obvious caveat that you must be on at least typescript 4.7 in order to use this
+feature.
+Your app will need to be on at least `sku@11.6.0` in order to use typescript 4.7.
+
+[instantiation expression]: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#instantiation-expressions
+
+### Toast types
+
+Braid does not expose a public `Toast` component.
+Instead, toasts are shown by calling the function returned from the [`useToast` hook][usetoast docs].
+If you wish to expose the parameters of this function on a component as props, then the types for those parameters can be derived from the `useToast` hook:
+
+```diff
+// Error in Braid v32
+-import { Toast } from 'braid-design-system/lib/components/useToast/ToastTypes';
+// Derive type from useToast instead
++import { useToast } from 'braid-design-system';
+
++type ShowToastFunction = ReturnType<typeof useToast>;
++type Toast = Parameters<ShowToastFunction>[0];
+
+interface MyComponentProps {
+  toast: Toast;
+}
+
+const MyComponent = ({ toast }: MyComponentProps) => {
+  const showToast = useToast();
+
+  return (
+    <Button onClick={() => showToast(toast)}}>
+      Show toast
+    </Button>
+  );
+};
+```
+
+[usetoast docs]: https://seek-oss.github.io/braid-design-system/components/useToast
+
+### Custom Icons
+
+If you use the private `useIcon` API in a custom icon within your app, there are a few options you
+should evaluate in order to migrate off this API:
+
+1. Evaluate the usage of your custom icon across your app. If your icon is only used at one specific size,
+1. Consider whether the icon could be replaced by an existing [Braid icon][braid icons]. Braid's iconography suite has expanded over the years, and your custom icon may have a suitable replacement available in Braid.
+1. If there is no suitable replacement for your custom icon available within Braid, then it could be worth adding to Braid. If your icon represents something generic, and not something product specific (e.g. product logo), then it could be a good candidate for Braid. Please reach out in the #braid-support Slack channel if you think your custom icon would fit into Braid's iconography suite.
```

Comment:
```suggestion
1. If there is no suitable replacement for your custom icon available within Braid, then it could be worth adding to Braid. If your icon represents something generic, and not something product specific (e.g. product logo), then it could be a good candidate for Braid. Please reach out in the [#braid-support] Slack channel if you think your custom icon would fit into Braid's iconography suite.
```